Formula/asdf.rb

```diff
@@ -4,30 +4,40 @@ class Asdf < Formula
   url "https://github.com/asdf-vm/asdf/archive/v0.8.0.tar.gz"
   sha256 "9b667ca135c194f38d823c62cc0dc3dbe00d7a9f60caa0c06ecb3047944eadfa"
   license "MIT"
-  revision 1
+  revision 2
   head "https://github.com/asdf-vm/asdf.git"
 
   bottle :unneeded
-
-  depends_on "autoconf"
-  depends_on "automake"
   depends_on "coreutils"
-  depends_on "libtool"
-  depends_on "libyaml"
-  depends_on "openssl@1.1"
-  depends_on "readline"
-  depends_on "unixodbc"
-
-  conflicts_with "homeshick",
-    because: "asdf and homeshick both install files in lib/commands"
+  depends_on "git"
 
   def install
     bash_completion.install "completions/asdf.bash"
     fish_completion.install "completions/asdf.fish"
     zsh_completion.install "completions/_asdf"
-    libexec.install "bin/private"
-    prefix.install Dir["*"]
-    touch prefix/"asdf_updates_disabled"
+    libexec.install Dir["*"]
+    touch libexec/"asdf_updates_disabled"
+    bin.write_exec_script (opt_libexec/"bin/asdf")
+    (prefix/"asdf.sh").write ". #{opt_libexec}/asdf.sh\n"
+    (prefix/"asdf.fish").write ". #{opt_libexec}/asdf.fish\n"
```

Comment:
Fish is not a POSIX shell and has [deprecated '.' support](https://fishshell.com/docs/current/cmds/source.html#cmd-source)

> . (a single period) is an alias for the source command. The use of . is deprecated in favour of source, and . will be removed in a future version of fish.

Reply:
@cho-m Thanks!

Reply:
Exports in Fish shell are [done through the `set -x` command (though may need `-g` if not already global variable)](https://fishshell.com/docs/current/cmds/set.html).

I am not entirely familiar with its idiosyncrasies since the spec is still in actively development compared to POSIX.

I think `PATH` is a special already-global variable.
Looking at [Fish tutorial](https://fishshell.com/docs/current/tutorial.html#path), prepending 2 directories is:
```fish
set PATH /usr/local/bin /usr/sbin $PATH
```


The part you quoted is due to Fish shell not supporting POSIX parameter expansion.
Thus, rather than `${ASDF_DATA_DIR:-$HOME/.asdf}`, they need a conditional check.

Reply:
@cho-m 

```shell
set PATH (                                                                          
  if test -n "$ASDF_DATA_DIR"                                                       
    echo $ASDF_DATA_DIR/shims                                                       
  else                                                                              
    echo $HOME/.asdf/shims                                                          
  end                                                                               
) $PATH  
```
Should be good enough for the caveat?

Reply:
I think it should work.

A side question is if the caveat is becoming too long. I don't know if there is a style recommendation.
Try to get a Homebrew maintainer to review everything again and decide the direction of what the caveat should say.
Hopefully should be last review to finalize PR merge.

Reply:
@cho-m Earlier suggestions I had was to put all this somewhere as a command, like `$ asdf setup` and have that as a `post_install` hook, but then again not sure if that also breaks Homebrew rules, @carlocab

I suppose the reason why this wasn't possible now was because `$ asdf setup` would just crash, but with this fix, it works out of the box. Maybe not without shims in $PATH, but `asdf` as a program works just fine.